lib/spack/spack/filesystem_view.py

```diff
@@ -40,6 +41,44 @@
 _projections_path = '.spack/projections.yaml'
 
 
+def view_symlink(src, dst, view, spec):
+    os.symlink(src, dst)
+
+
+def view_hardlink(src, dst, view, spec):
+    os.link(src, dst)
+
+
+def view_copy(src, dst, view, spec):
+    shutil.copyfile(src, dst)
+    if spec:
+        # Not metadata, we have to relocate it
+
+        # What type of file are we relocating
+        relocate_method = spack.relocate.relocate_text_bin \
+            if spack.relocate.is_binary(dst) else spack.relocate.relocate_text
+
+        # Get information on where to relocate from/to
+        prefix_to_prefix = dict(
```

Comment:
Wondering if this name:
```suggestion
        prefix_to_projection = dict(
```
describes better what we are doing.

Reply:
It does. The other was to match the function signature of the method I call with it as an argument, but I'll change it now that I'm using kwargs (as requested below).